portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,299 @@
+ *    Its keys contain the name of each contact section, and its values are
+ *    either a single or an array of contact data.
+ */
+function loadContactData(contactData) {
+  const contactsContainer = document.querySelector('header > .contact-container');
+  const entries = createContactEntries(contactData);
+
+  contactsContainer.append(...entries);
+}
+
+/**
+ * This function converts each section in the given contact data to 
+ * an html node based on the "contact-entry" template.
+ * 
+ * @param {!object} contactData Dictionary containing the contact data to be shown. 
+ *    Its keys contain the name of each contact section, and its values are
+ *    either a single or an array of contact data.
+ * 
+ * @returns {!Array} The array of "contact-entry" html nodes.
+ */
+function createContactEntries(contactData) {
+  const contactEntryTemplate = document.querySelector('#contact-entry');
+  const entries = Object.entries(contactData).map(
+    ([title, data]) => loadContactEntry(contactEntryTemplate.content.cloneNode(true), title, data)
+  );
+
+  return entries;
+}
+
+/**
+ * This function populates an empty "contact-entry" html node given the contact data. 
+ * 
+ * @param {!DocumentFragment} entry The empty contact-entry html node.
+ * @param {!string} title The title of the current contact-entry.
+ * @param {!(string | {text: string, link: string} | Array<string | {text: string, link: string}>)} data 
+ *    A single or an array of contact data, which could be a string or an object with text and a link.
+ * 
+ * @returns {!DocumentFragment} The populated entry.
+ */
+function loadContactEntry(entry, title, data) {
+  entry.querySelector('.title').append(title);
+
+  if (Array.isArray(data)) {
+    valueDivs = data.map(createContactEntryContent);
+    entry.querySelector('.content').append(...valueDivs);
+  } else {
+    const div = createContactEntryContent(data);
+    entry.querySelector('.content').append(div);
+  }
+
+  return entry;
+}
+
+/**
+ * This function creates either a div or an anchor element,
+ * based on the provided data.
+ * 
+ * @param {!(string | {text: string, link: string})} data 
+ *    Contact information to be used in the created element
+ * 
+ * @returns {!HTMLElement} The element containing contact information
+ */
+function createContactEntryContent(data) {
+  const node = document.createElement('div');;
+
+  if (typeof data === 'object') {
+    const { text, link } = data;
+
+    const a = document.createElement('a');
+    a.href = link;
+    a.append(text);
+
+    node.append(a);
+  } else {
+    node.append(data);
+  }
+
+  return node;
+}
+
+/**
+ * Processes and appends the given experience section information on the website body.
+ * 
+ * @param {!object} sectionData Dictionary containing the experience data to be shown. 
+ *    Its keys contain the name of each experience section, and its values are
+ *    an array of objects describing individual experiences.
+ */
+function loadSectionData(sectionData) {
```

Comment:
The name here is a bit misleading, because it's not loading but creating and appending elements. Maybe `buildSectionData()` ?

Reply:
Hmm... How would you define "loading"?

My thought process originally was:

- Functions that interact with DOM and return nothing: loadSomething
- Functions that create some DOM element based on parameters and return it: createSomething

(I actually ignored this myself by accident and named functions that populate individual nodes "loadSomething" as well. Maybe I could rename them to "populateSomething")

Reply:
In JS world, `build` is used when constructing DOM elements, and `render` is used when appending to the DOM. `load` is usually reserved for when we fetch something from a remote server.

Reply:
Sounds good! Working on it

Reply:
Done in latest changes. How about now?